scripts/update/update_genes.py

```diff
@@ -0,0 +1,571 @@
+                wr.write(f"ERROR: more than one stable_id for {gene_symbol}: {gene_symbol_2_stable_id[gene_symbol]}\n")
+            else:
+                unique_gene_symbol_2_stable_id[gene_symbol] = list(gene_symbol_2_stable_id[gene_symbol])[0]
+                wr_genes.write(f"{unique_gene_symbol_2_stable_id[gene_symbol]}\t{gene_symbol}\n")
+                # save the ensembl id as key for easier comparison later
+                unique_stable_id_2_gene_symbol[unique_gene_symbol_2_stable_id[gene_symbol]] = {
+                    "gene_symbol": gene_symbol,
+                    "chr": gene_symbol_details[gene_symbol]["chr"],
+                    "start": gene_symbol_details[gene_symbol]["start"],
+                    "end": gene_symbol_details[gene_symbol]["end"],
+                    "strand": gene_symbol_details[gene_symbol]["strand"]
+                }
+
+    return unique_stable_id_2_gene_symbol
+
+def get_g2p_genes(db_host, db_port, db_name, user, password):
+    g2p_genes = {}
+    g2p_genes_by_symbol = {}
+
+    sql =   """
+                SELECT l.name, li.identifier, l.id, l.start, l.end, seq.name FROM locus l
+                LEFT JOIN locus_identifier li ON li.locus_id = l.id
+                LEFT JOIN source s ON s.id = li.source_id
+                LEFT JOIN sequence seq ON seq.id = l.sequence_id
+                WHERE s.name = 'Ensembl'
+            """
+
+    db = MySQLdb.connect(host=db_host, port=db_port, user=user, passwd=password, db=db_name)
+    cursor = db.cursor()
+    cursor.execute(sql)
+    data = cursor.fetchall()
+    for row in data:
+        g2p_genes[row[1]] = {
+            "gene_symbol": row[0],
+            "locus_id": row[2],
+            "locus_start": row[3],
+            "locus_end": row[4],
+            "locus_chr": row[5],
+        }
+        g2p_genes_by_symbol[row[0]] = row[1]
+
+    db.close()
+
+    return g2p_genes, g2p_genes_by_symbol
+
+def get_g2p_genes_hgnc(db_host, db_port, db_name, user, password):
+    """
+        Dump the full G2P gene set.
+        Returns a dictionary where the key is the gene symbol.
+        Called by: update_xrefs()
+    """
+    g2p_genes_by_symbol = {}
+
+    sql =   """
+                SELECT l.name, li.identifier, l.id, la.value FROM locus l
+                LEFT JOIN locus_identifier li ON li.locus_id = l.id
+                LEFT JOIN locus_attrib la ON la.locus_id = l.id
+                LEFT JOIN source s ON s.id = li.source_id
+                WHERE s.name = 'Ensembl'
+            """
+
+    sql_ids =   """
+                    SELECT l.name, li.identifier FROM locus l
+                    LEFT JOIN locus_identifier li ON li.locus_id = l.id
+                """
+
+    db = MySQLdb.connect(host=db_host, port=db_port, user=user, passwd=password, db=db_name)
+    cursor = db.cursor()
+    cursor.execute(sql)
+    data = cursor.fetchall()
+    for row in data:
+        if row[0] not in g2p_genes_by_symbol:
```

Comment:
Lets use defaultdict here - so we do not have to check for ```if row[0] not in g2p_genes_by_symbol```

Code - 
```
from collections import defaultdict
g2p_genes_by_symbol = defaultdict(lambda: {
"stable_id" : None,
"locus_id" : None,
"synonyms" : []
})

for row in data:
symbol = row[0]
g2p_genes_by_symbol[symbol]["stable_id"] = row[1],
g2p_genes_by_symbol[symbol]["locus_id"] = row[2],
g2p_genes_by_symbol[symbol]["synonyms'].append(row[3]
```

Reply:
I understand that defaultdict can simplify the dictionary creation but personally, I find the code easier to read and maintain without it. Also, defaultdict is going to overwrite the "stable_id" and "locus_id" each time a "synonym" is appended which can be avoided.